modules/Virtualizer_module/src/VirtualizerModule.cpp

```diff
@@ -218,21 +392,75 @@ bool VirtualizerModule::updateModule()
                  << tmpSpeedDirection;
         return false;
     }
-    // velocity_factor=2;
-    //   double x = speedData * cos(angulareError) * velocity_factor;
-    //   double y = speedData * sin(angulareError) * velocity_factor;
+
     double x = speedDirection * m_scale_X * speedData;
-    double y = m_scale_Y * angulareError;
+    double y = 0;
+    if (m_useRingVelocity)
+    {
+        double newVelocity = Angles::shortestAngularDistance(m_oldPlayerYaw, playerYaw)/getPeriod();
+        double filteredVelocity = threshold(filteredRingVelocity(newVelocity), m_velocityDeadzone);
+        y = -m_velocityScaling * filteredVelocity; //The ring has the z axis pointing downward
+    }
+    else
+    {
+        // get the robot orientation
+        yarp::sig::Vector* tmp = m_robotOrientationPort.read(false);
+        if (tmp != NULL)
+        {
+            auto vector = *tmp;
+            m_robotYaw = -Angles::normalizeAngle(vector[0]);
+        }
+
+        // error between the robot orientation and the player orientation
+        double angularError = threshold(Angles::shortestAngularDistance(m_robotYaw, playerYaw));
+        y = -m_scale_Y * angularError; // because the virtualizer orientation value is CCW, therefore we put "-" to
+        // make it CW, same as the robot world.
+    }
+
+    if (m_useHeadForTurning)
+    {
+        if (std::fabs(speedData) > m_isMovingDeadzone)
```

Comment:
> I think it is a good idea that you have added the dead zone for the speed data.
> It happens some times that the user by mistake moves or sensory errors causes a small speed value returned from the virtualizer

Note that the meaning of that parameter is different. In particular, that is supposed to be a little larger than a normal threshold since we want to be sure that the person is moving before using the head to control the direction. It is used to activate a feature, rather than filtering the person's movements. Probably, instead of `m_isMovingDeadzone` should be `m_isMovingThreshold`. What do you think?

I guess that for filtering the velocity there is the ``deadzone``: https://github.com/S-Dafarra/walking-teleoperation/blob/0afda2db52b25c00ab3a4dadfa4c458a30c3b20b/app/robots/iCubGenova09/virtualizerConfig.ini#L5 that does not seem to be used to filter the ``x`` velocity: https://github.com/S-Dafarra/walking-teleoperation/blob/0afda2db52b25c00ab3a4dadfa4c458a30c3b20b/modules/Virtualizer_module/src/VirtualizerModule.cpp#L396

Reply:
>I guess that for filtering the velocity there is the deadzone: https://github.com/S-Dafarra/walking-teleoperation/blob/0afda2db52b25c00ab3a4dadfa4c458a30c3b20b/app/robots/iCubGenova09/virtualizerConfig.ini#L5 that does not seem to be used to filter the x velocity: https://github.com/S-Dafarra/walking-teleoperation/blob/0afda2db52b25c00ab3a4dadfa4c458a30c3b20b/modules/Virtualizer_module/src/VirtualizerModule.cpp#L396

The `deadzone` variable you referred to is used for angular error threshold in https://github.com/S-Dafarra/walking-teleoperation/blob/0afda2db52b25c00ab3a4dadfa4c458a30c3b20b/modules/Virtualizer_module/src/VirtualizerModule.cpp#L415



> Note that the meaning of that parameter is different. In particular, that is supposed to be a little larger than a normal threshold since we want to be sure that the person is moving before using the head to control the direction. It is used to activate a feature, rather than filtering the person's movements. Probably, instead of `m_isMovingDeadzone` should be `m_isMovingThreshold`. What do you think?

Yes, I understood you use this introduced threshold to understand if the user is to start moving (i.e., walking) or not.
That's why I suggested bringing this variable before in order to avoid that the undesirable behavior I was mentioning. If you think, the threshold you are setting here is high (and is only for the usage of neck yaw angle retargeting) and is better to use a smaller `deadzone` for the speed, we may introduce another variable for that. But I wanted to avoid it and make a consensus on one deadzone/threshold for this. 
About the naming, what you suggested it OK for me. But if we merge to one variable or for the speed deadzone we may use `speed_deadzone` as the name!

Reply:
Ok, fine for me! I will introduce the ``speed_deadzone`` parameter! At this point, I would also rename ``deadzone`` as it is way too generic. What do you think?

Reply:
Agree!

Reply:
> Ok, fine for me! I will introduce the `speed_deadzone` parameter! At this point, I would also rename `deadzone` as it is way too generic. What do you think?

yes, it is better to have it more specific. Maybe then while you are doing this addition, you can also remove the overloaded `threshold` function https://github.com/robotology/walking-teleoperation/pull/54/files#diff-340d6ca0ad3b2f551f4052b6c7fc90549549d6865a27a276bb5e83b86d01c894L62 with the default value of `m_deadzone`!
Thanks.

Reply:
- Added the ``speed_deadzone``
- Removed ``threshold`` method without the input deadzone value
- Removed ``is_moving_tolerance``
in https://github.com/robotology/walking-teleoperation/pull/54/commits/f26644d52d9917669d60e707e20c3bcdb8a08914